File: ProyectoV2022/Probando/Entities/Usuario.cs
using ServicioMedico.Data;
using System.ComponentModel.DataAnnotations;

namespace ServicioMedico.Entities
{
    public class Usuario : IEntity
    {
        [Key]
        public long Id { get; set; }

        [Required]
        [StringLength(15)]
        public string nombre { get; set; }

        [Required]
        [StringLength(25)]
        public string apellidos { get; set; }

        [Required]
        [StringLength(18)]
        public string usuario { get; set; }
        
        [Required]
        [StringLength(20)]
        public string clave { get; set; }

    }
}

File: ProyectoSpring/src/main/java/net/acc/customproyect/entities/Usuario.java
package net.acc.customproyect.entities;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Table;

@Entity
@Table(name = "USUARIO")
@Inheritance(strategy = InheritanceType.JOINED)
public class Usuario {

	@Id
	@GeneratedValue(strategy= GenerationType.SEQUENCE)
	@Column(name = "usuario_id")
	private Long usuarioId;
	@Column(name = "nombre")
	private String nombre;
	@Column(name = "apellidos")
	private String apellidos;
	@Column(name = "email")
	private String email;
	

	public Usuario() {
		super();
	}

	public Usuario(String nombre, String apellidos, String email, String clave) {
		super();
		this.nombre = nombre;
		this.apellidos = apellidos;
		this.email = email;
	}
	
	

	public Long getUsuarioId() {
		return usuarioId;
	}

	public void setUsuarioId(Long usuarioId) {
		this.usuarioId = usuarioId;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getApellidos() {
		return apellidos;
	}

	public void setApellidos(String apellidos) {
		this.apellidos = apellidos;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

}
